rollouts/docs/running-locally.md

```diff
@@ -0,0 +1,176 @@
+  confirmed to work on Linux)
+* [go 1.19](https://go.dev/dl/) or newer
+* `make`
+* Access to GKE clusters.
+
+## Cluster Setup
+
+### Creating a management cluster with KCC running
+First, you must create a config-controller cluster. You can follow the instructions on the 
+[config-controller-setup guide](https://cloud.google.com/anthos-config-management/docs/how-to/config-controller-setup) to create a config-controller cluster. Make sure your kubeconfig is 
+connected to this cluster.
+
+### Provisioning child clusters
+The next step will be to provision new child clusters. This example names the child cluster `gke-n`;
+you can replace `gke-n` with whatever name you want for your child cluster. 
+
+```sh
+# first step is to create a deployable instance of the package
+kpt pkg get git@github.com:droot/kpt-packages.git/cluster@main gke-n --for-deployment
+
+# now render the package
+kpt fn render
+
+# assuming your kubectl is configured to talk to a management cluster with KCC running (the previous step).
+# provision the cluster using following commands
+kpt live init gke-n
+kpt live apply gke-n
+```
+
+You can repeat the above steps to create as many child clusters as you want.
+
+### Setting up kubeconfig
+Next, we will configure kubeconfig to be able to talk to each cluster individually.
+
+```sh
+# once kpt live status is showing current for a cluster package
+# run the following
+KUBECONFIG=~/.kube/gke-n gcloud container clusters get-credentials gke-n --region us-west1
+
+# now ~/.kube/gke-n file has been updated with the credentials for gke-n cluster
+# verify if this is working
+
+KUBECONFIG=~/.kube/gke-n kubectl get pods -n kube-system
+```
+
+### Set up Config Sync
+Rollouts requires a git syncer installed on child clusters. Currently, only Config Sync is supported. 
+To install Config Sync on your child clusters:
+
+```sh
+kpt pkg get git@github.com:droot/kpt-packages.git/config-sync@main config-sync
+cd config-sync/
+KUBECONFIG=~/.kube/gke-n kubectl apply -f manifests.yaml
+```
+
+## Running Rollouts locally
+
+Clone this repository into `${GOPATH}/src/github.com/GoogleContainerTools/kpt`.
+
+```sh
+git clone https://github.com/GoogleContainerTools/kpt.git "${GOPATH}/src/github.com/GoogleContainerTools/kpt"
+```
+
+Enter the rollouts directory:
+
+```
+cd rollouts
+```
+
+Download dependencies:
+
+```sh
+make tidy
+```
+
+Assuming your kubeconfig is configured to talk to the management cluster with KCC installed (that you created
+in the above steps), apply the manifests:
+
+```sh
+make install
+```
+
+Confirm the CRDs are installed:
+
+```sh
+kubectl api-resources | grep gitops.kpt.dev
+
+progressiverolloutstrategies  gitops.kpt.dev/v1alpha1  true  ProgressiveRolloutStrategy
+remotesyncs                   gitops.kpt.dev/v1alpha1  true  RemoteSync
+rollouts                      gitops.kpt.dev/v1alpha1  true  Rollout
+```
+
+Now you are ready to run the controller locally:
+
+```sh
+go run main.go
```

Comment:
I generally recommend setting `KUBECONFIG=~/.kube/admin-cluster go run main.go` to avoid running with default kubectl context.

Reply:
Updated